src/vs/platform/terminal/common/terminal.ts

```diff
@@ -391,7 +393,7 @@ export interface IShellLaunchConfig {
 	 * The codicon ID to use for this terminal. If not specified it will use the default fallback
 	 * icon.
 	 */
-	icon?: string;
+	iconPath?: ThemeIcon | Codicon | string | URI | { light: URI, dark: URI };
```

Comment:
We could make a union type here for convenience internally, just as long as we keep the one in vscode.proposed.d.ts the full type.

```
type TerminalIcon = ThemeIcon | URI | { light: URI, dark: URI };
```

Reply:
decided against this for the reason you mentioned yesterday - I like being able to see on hover the types I can set iconPath to